### ISSUE TITLE
Installation problem 

### ISSUE THREAD

**Issue by lgmendes** [Seilim/CaffeLink#1]
Hi,

Can you help me building the CaffeLink. When I try I found the following error:
![caffelink_problem](https://cloud.githubusercontent.com/assets/4982469/10483196/e0df3a04-7275-11e5-893a-ef86af6a0deb.jpg)

I have build the Caffe without GPU.
The edited make file is 
![makefile_caffelink](https://cloud.githubusercontent.com/assets/4982469/10483363/ff83bd6c-7276-11e5-99f1-022b23a5dd27.jpg)

Thank you

All my best

**Comment by Seilim** [comment]
Hi,
this should be solved by adding `CPU_ONLY` option to makefile: `CXXFLAGS=-c -Wall -DCPU_ONLY`. Now even CaffeLink will be compiled without GPU support. Also, make sure you initialize it with `"UseGPU"->False` in Mathematica.

**Comment by lgmendes** [comment]
Thanks!:+1: 
Now I have another problem:
![caffelink_error2](https://cloud.githubusercontent.com/assets/4982469/10487546/5c1fbfb6-728c-11e5-83f7-456d5aa4ea46.jpg)

can you help me ?

Thank you

**Comment by Seilim** [comment]
Yeah, that is because you have Caffe rc2 or newer. And I apparently did not push an update. I think I did not feel like changing examples and fixing broken parsing of caffe.proto back then. Oh well, I'm going to check what I have and probably create a new branch for now.

**Comment by lgmendes** [comment]
Thank you!

All  my best

**Comment by lgmendes** [comment]
Thank you!!

**Comment by Seilim** [comment]
Ok, [Caffe rc2](/Seilim/CaffeLink/tree/caffe_rc2) branch is ready. Net definition generator and examples are still rc1 though. More info in [read me](/Seilim/CaffeLink/tree/caffe_rc2/README.md).

**Comment by lgmendes** [comment]
Thank you!! :+1: 
That problem is solved. However I found another problem:
![caffelink_rc2_problem](https://cloud.githubusercontent.com/assets/4982469/10510135/0ba85cee-7329-11e5-8fc7-ad06b4b27050.jpg)

I will try to figure out the solution

**Comment by Seilim** [comment]
Are you on Os x? There are a bit different dependencies. If so, please try:
`make -f Makefile.osx`

**Comment by lgmendes** [comment]
Problem solved. :+1: 
 Makefile.osx works well. In my case I had only to change to the libc++ (Mac OS 10.9).

The libcaffeLink.dylib was generated and I copy to a folder on $LibraryPath.
At this point I have problems in the initialisation of the CaffeLink in Mathematica:
![caffelink_pro](https://cloud.githubusercontent.com/assets/4982469/10513655/f4c3bc90-7340-11e5-8882-ab7ebedac615.jpg)

**Comment by Seilim** [comment]
Now Mathematica has probably trouble finding some libraries CaffeLink needs. `LibraryLink```$LibraryError` might tell you what it is. I guess it is `libcaffe.dylib`. I am not familiar with osx so I can only give you this [hint](http://mathematica.stackexchange.com/a/34018).

**Comment by lgmendes** [comment]
:+1: :+1: 
install_name_tool -id full/path/to/lib/to/load libfile.dylib  
solve the problem. ImageNet.nb and libLink-test.nb work without any problem.

Thank you again!
